druid/src/widget/widget.rs

```diff
@@ -46,6 +46,11 @@ use super::prelude::*;
 /// [`LifeCycleCtx::widget_id`]: struct.LifeCycleCtx.html#method.widget_id
 /// [`WidgetExt::with_id`]: trait.WidgetExt.html#method.with_id
 /// [`IdentityWrapper`]: widget/struct.IdentityWrapper.html
+//
+// The widget id does not necessarily identify only a single widget
```

Comment:
This is true, but might be confusing.

`WidgetPod` is not really a widget. Or: it _can_ be, but that's sort of also confusing.

The best way to think of `WidgetPod` is that it is the base class of `Widget`. In certain instances, we can have a widget that has another widget as a child, without a `WidgetPod`; but I would also almost think of this as a sort of subclassing, a way of modifying the child's behaviour in some way; the two widgets are really acting as a single until.

In a sense this would be more clear if `WidgetPod` was named `Widget`, and `Widget` was named `Component`; conceptually, it can be helpful to think of a 'widget' as being all of the `Widget`s between one `WidgetPod` and it's closest descendent `WidgetPod`.

Reply:
Yeah my understanding about this is getting refined every day. I was thinking about proposing renaming `WidgetPod` to `Widget` just earlier today. I think this is something worth discussing in length in zulip some day.

The origin story of this comment is me finding that the window base state copies the root `WidgetPod` id which means there's a duplicated id in the event path. Maybe I'll just change this comment here to mention only that, or maybe even move that comment to window instead.

Reply:
yea, that's slightly hacky; iirc the root window's basestate really just exists because we need _some_ initial basestate in order to create the various contexts; we do check some things on it after a cycle, but we could just as easily check those on the root widget.